handlers/handlers.go

```diff
@@ -246,16 +245,20 @@ func HandleHomeRequest(
 
 		parsedTemplate.ExecuteTemplate(responseWriter, baseTemplateName, userId)
 	default:
-		respondWithMethodNotAllowed(responseWriter, []string{http.MethodGet})
+		respondWithMethodNotAllowed(responseWriter, http.MethodGet)
 	}
 }
 
 // PRIVATE
 
 func respondWithMethodNotAllowed(
 	responseWriter http.ResponseWriter,
-	allowedMethods []string,
+	allowedMethods ...string,
 ) {
+	if allowedMethods == nil {
+		return
```

Comment:
Shouldn't this be an error? when would it be reasonable to call respond with method not allowed, with no inputs.

Reply:
I'm in here. Lurking in your code reviews

You could make this a compiler error by making the signature

```go
func respondWithMethodNotAllowed(
  responseWriter http.ResponseWriter, 
  allowedMethod string, 
  additionalMethods  ...string)
```

then maybe the first line of this method is to concatenate the required string with the  optional strings